src/pymor/algorithms/line_search.py

```diff
@@ -56,15 +60,25 @@ def armijo(f, starting_point, direction, grad=None, initial_value=None,
     slope = 0.0
 
     # Compute slope if gradient is provided
-    if grad:
-        slope = min(grad.inner(direction), 0.0)
+    if grad is not None:
+        if isinstance(grad, np.ndarray):
```

Comment:
I think it's ok to allow both VectorArrays an numpy arrays here, but the docstring needs to be updated.